### PR TITLE
[BugFix] fix tablet writer add chunk when larger than 2GB

### PR DESCRIPTION
**Pull request by luohaha**
## Why I'm doing:
There two issues now when handle >2GB chunk:

1. `starrocks.PTabletWriterAddChunksRequest exceeded maximum protobuf size of 2GB: 4194416177`
```
void serialize_to_iobuf(const T& proto_obj, butil::IOBuf* iobuf) {
    butil::IOBuf tmp_iobuf;
    butil::IOBufAsZeroCopyOutputStream wrapper(&tmp_iobuf);
    proto_obj.SerializeToZeroCopyStream(&wrapper);
```

`SerializeToZeroCopyStream` will fail when handle >2GB chunk.

2. In previous pr #40229, It introduced a bug about missing set `http` protocol for BRPC client.
```
 input_messenger.cpp:96] A message from 172.26.94.64:39672(protocol=baidu_std) is bigger than 2147483648 bytes, the connection will be closed. Set max_body_size to allow bigger messages
```

## What I'm doing:
Fix these two issues:
1. Seperate chunk from protobuf, so `SerializeToZeroCopyStream` won't fail because of >2GB serialize.
2. Set http protocol for `brpc::ChannelOptions` when handle >2GB RPC.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by chaoyli — DISMISSED** [review]

File: be/src/util/internal_service_recoverable_stub.cpp [hunk 1]
```diff
@@ -49,13 +49,18 @@ PInternalService_RecoverableStub::PInternalService_RecoverableStub(const butil::
 
 PInternalService_RecoverableStub::~PInternalService_RecoverableStub() = default;
 
-Status PInternalService_RecoverableStub::reset_channel() {
+Status PInternalService_RecoverableStub::reset_channel(const std::string& protocal) {
```
Comment (wyb) [inline be/src/util/internal_service_recoverable_stub.cpp:52]:
```suggestion
Status PInternalService_RecoverableStub::reset_channel(const std::string& protocol) {
```

File: be/src/service/service_be/internal_service.cpp [hunk 2]
```diff
@@ -98,22 +98,50 @@ void BackendInternalServiceImpl<T>::tablet_writer_add_chunks(google::protobuf::R
...
+        }
+        auto size = iobuf.cutn(chunk->mutable_data(), chunk->data_size());
+        if (size != chunk->data_size()) {
+            LOG(ERROR) << fmt::format("iobuf read {} != expected {}.", size, chunk->data_size());
+            return false;
+        }
+    } else if constexpr (std::is_same<T, PTabletWriterAddChunksRequest>::value) {
+        for (int i = 0; i < proto_obj->requests_size(); i++) {
+            auto chunk = proto_obj->mutable_requests(i)->mutable_chunk();
+            if (iobuf.size() < chunk->data_size()) {
```
Comment (wyb) [inline be/src/service/service_be/internal_service.cpp:133]: iobuf left size?

File: be/src/exec/tablet_sink_index_channel.cpp [hunk 3]
```diff
@@ -548,14 +548,43 @@ Status NodeChannel::_filter_indexes_with_where_expr(Chunk* input, const std::vec
...
+// | protobuf (without chunk) |
+// | chunk (1) |
+// | chunk (2) |
+// | chunk (...) |
+// | chunk (N) |
 template <typename T>
-void serialize_to_iobuf(const T& proto_obj, butil::IOBuf* iobuf) {
-    butil::IOBuf tmp_iobuf;
-    butil::IOBufAsZeroCopyOutputStream wrapper(&tmp_iobuf);
+void serialize_to_iobuf(T& proto_obj, butil::IOBuf* iobuf) {
```
Comment (wyb) [inline be/src/exec/tablet_sink_index_channel.cpp:561]: compatibility when upgrading?

File: be/src/service/service_be/internal_service.cpp (hunk 2 above)
Comment (luohaha) [reply be/src/service/service_be/internal_service.cpp:133]: iobuf.size() is already left size after `cutn`

File: be/src/util/internal_service_recoverable_stub.cpp (hunk 1 above)
Comment (luohaha) [reply be/src/util/internal_service_recoverable_stub.cpp]: done

**Review by wyb — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by chaoyli** [comment]
@mergify backport branch-3.3

**Pull request merged**